File: blacklist-backend/app/schemas/order.py
"""
订单相关模式
"""
from pydantic import BaseModel, Field, validator
from typing import Optional, List, Union
from datetime import datetime
from decimal import Decimal
from app.models.order import OrderStatus


class OrderBase(BaseModel):
    """订单基础模式"""
    group_id: Optional[int] = None
    group_tour_number: Optional[str] = None
    orderer: Optional[str] = None
    member_remarks: Optional[str] = None
    payment_time: Optional[datetime] = None
    group_leader_remarks: Optional[str] = None
    product: Optional[str] = None
    order_amount: Optional[Decimal] = None
    refund_amount: Optional[Decimal] = 0
    order_status: Optional[Union[OrderStatus, str]] = None
    pickup_point: Optional[str] = None
    consignee: Optional[str] = None
    contact_phone: Optional[str] = None
    detailed_address: Optional[str] = None
    is_blacklist_checked: Optional[str] = "no"
    blacklist_risk_level: Optional[str] = None
    blacklist_match_info: Optional[str] = None
    blacklist_match_details: Optional[str] = None
    
    @validator('order_status', pre=True)
    def validate_order_status(cls, v):
        if v is None or v == "" or v == "":
            return None
        if isinstance(v, str):
            # 尝试转换为OrderStatus枚举
            try:
                return OrderStatus(v.lower())
            except ValueError:
                return None
        return v


class OrderCreate(OrderBase):
    """创建订单模式"""
    pass


class OrderUpdate(OrderBase):
    """更新订单模式"""
    pass


class OrderInDB(OrderBase):
    """数据库中的订单模式"""
    id: int
    created_at: datetime
    updated_at: datetime
    is_active: bool

    class Config:
        from_attributes = True


class Order(OrderInDB):
    """订单响应模式"""
    pass


class OrderListResponse(BaseModel):
    """订单列表响应模式"""
    data: List[OrderInDB]
    total: int
    page: int
    size: int


class OrderSearchParams(BaseModel):
    """订单搜索参数"""
    group_id: Optional[int] = None
    group_tour_number: Optional[str] = None
    orderer: Optional[str] = None
    contact_phone: Optional[str] = None
    order_status: Optional[Union[OrderStatus, str]] = None
    is_blacklist_checked: Optional[str] = None
    payment_time_start: Optional[datetime] = None
    payment_time_end: Optional[datetime] = None
    skip: int = 0
    limit: int = 20
    
    @validator('order_status', pre=True)
    def validate_order_status(cls, v):
        if v is None or v == "" or v == "":
            return None
        if isinstance(v, str):
            # 尝试转换为OrderStatus枚举
            try:
                return OrderStatus(v.lower())
            except ValueError:
                return None
        return v


class ExcelUploadResponse(BaseModel):
    """Excel上传响应模式"""
    success: bool
    message: str
    imported_count: int
    failed_count: int
    errors: List[str] = []


class BlacklistCheckResponse(BaseModel):
    """黑名单检测响应模式"""
    is_blacklist: bool
    risk_level: Optional[str] = None
    match_info: Optional[str] = None
    matched_blacklist_ids: List[int] = []

File: blacklist-frontend/src/api/order.ts
/**
 * 订单相关API
 */
import request from './request'

export interface OrderSearchParams {
  group_tour_number?: string
  orderer?: string
  contact_phone?: string
  order_status?: string
  is_blacklist_checked?: string
  payment_time_start?: string
  payment_time_end?: string
  skip?: number
  limit?: number
}

export interface OrderCreateData {
  group_tour_number?: string
  orderer?: string
  member_remarks?: string
  payment_time?: string
  group_leader_remarks?: string
  product?: string
  category?: string
  quantity?: number
  order_amount?: number
  refund_amount?: number
  order_status?: string
  pickup_point?: string
  consignee?: string
  contact_phone?: string
  detailed_address?: string
}

export interface OrderUpdateData extends OrderCreateData {}

export interface OrderListResponse {
  data: any[]
  total: number
  page: number
  size: number
}

export interface ExcelUploadResponse {
  success: boolean
  message: string
  imported_count: number
  failed_count: number
  errors: string[]
}

export interface BlacklistCheckResponse {
  is_blacklist: boolean
  risk_level?: string
  match_info?: string
  matched_blacklist_ids: number[]
}

export const orderApi = {
  // 获取订单列表
  getOrderList: (params: OrderSearchParams): Promise<OrderListResponse> => {
    return request.post('/orders/list', params)
  },

  // 获取单个订单
  getOrder: (id: number): Promise<any> => {
    return request.post('/orders/detail', { order_id: id })
  },

  // 创建订单
  createOrder: (data: OrderCreateData): Promise<any> => {
    return request.post('/orders/create', data)
  },

  // 更新订单
  updateOrder: (id: number, data: OrderUpdateData): Promise<any> => {
    return request.post('/orders/update', { order_id: id, ...data })
  },

  // 删除订单
  deleteOrder: (id: number): Promise<any> => {
    return request.post('/orders/delete', { order_id: id })
  },

  // 批量删除订单
  batchDeleteOrders: (ids: number[]): Promise<any> => {
    return request.post('/orders/batch-delete', { ids })
  },

  // 上传Excel文件
  uploadExcel: (file: File): Promise<ExcelUploadResponse> => {
    const formData = new FormData()
    formData.append('file', file)
    return request.post('/orders/upload-excel', formData, {
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    })
  },

  // 检测黑名单
  checkBlacklist: (id: number): Promise<BlacklistCheckResponse> => {
    return request.post(`/orders/${id}/check-blacklist`)
  },

  // 批量检测黑名单
  batchCheckBlacklist: (ids: number[]): Promise<{ results: any[] }> => {
    return request.post('/orders/batch-check-blacklist', { order_ids: ids })
  },

  // 导出订单
  exportOrders: (params: any): Promise<any> => {
    return request.post('/orders/export', params, {
      responseType: 'blob'
    })
  }
}
